vault/resource_aws_secret_backend_role.go

```diff
@@ -113,9 +123,16 @@ func awsSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) error {
 
 	credentialType := d.Get("credential_type").(string)
 
+	userPath := d.Get("user_path").(string)
+
+	permissionBoundaryArn := d.Get("permissions_boundary_arn").(string)
+
 	data := map[string]interface{}{
 		"credential_type": credentialType,
 	}
+	if d.HasChange("permissions_boundary_arn") {
```

Comment:
It might be a good idea to log a warning when the boundary ARN is configured and the `credential_type` is not `iam_user`.